stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -293,23 +288,25 @@ private void runInternal() throws Exception {
       CommonUtils.sleepMs(waitMs);
       mStartBarrierPassed = true;
 
-      for (int i = 0; i < mFilesPath.size(); i++) {
-        mCurrentOffset = 0;
-        String filePath = mFilesPath.get(i);
-        while (!Thread.currentThread().isInterrupted()) {
-          if (isRead && CommonUtils.getCurrentMs() > mContext.getEndMs()) {
-            closeInStream();
-            return;
-          }
-          long ioBytes = applyOperation(filePath);
-
-          // Start recording after the warmup
-          if (CommonUtils.getCurrentMs() > recordMs) {
-            if (ioBytes > 0) {
-              mThreadCountResult.incrementIOBytes(ioBytes);
-            } else {
-              // Done reading/writing one file
-              break;
+      for (int dirId = mThreadId; dirId < mParameters.mNumDirs; dirId += mParameters.mThreads) {
+        for (int fileId = 0; fileId < mParameters.mNumFilesPerDir; fileId++) {
+          mCurrentOffset = 0;
+          String filePath = mParameters.mLocalPath + "/" + dirId + "/" + fileId;
+          while (!Thread.currentThread().isInterrupted()) {
+            if (isRead && CommonUtils.getCurrentMs() > mContext.getEndMs()) {
+              closeInStream();
+              return;
+            }
+            long ioBytes = applyOperation(filePath);
+
+            // Start recording after the warmup
+            if (CommonUtils.getCurrentMs() > recordMs) {
+              if (ioBytes > 0) {
+                mThreadCountResult.incrementIOBytes(ioBytes);
+              } else {
+                // Done reading/writing one file
+                break;
```

Comment:
if (CommonUtils.getCurrentMs() < recordMs && ioBytes <= 0 ) { we should also break! we also done reading/writing one file
Move the break logic after long ioBytes = applyOperation(filePath);